### DISCUSSION TITLE
Implementation of `last_plot()` as closure: just old-fashioned or does that have other advantages?

### DISCUSSION THREAD

**Discussion by yjunechoe** [ggplot2-extenders/ggplot-extension-club#43] [General]
Ever since finding out that [`last_plot()`](https://github.com/tidyverse/ggplot2/blob/main/R/plot-last.R) is implemented as a closure, I've always wondered how that relates to the recent-ish trend in r-lib/tidyverse to use package environments (e.g. `the`) for stateful behaviors.

Is the implementation of `last_plot()` as closure simply a product of older times before the design principle of using package environments was fully flushed out? Or are there meaningful differences (advantage or disadvantage) to using closure vs. tracking state in a package env?

**Comment by teunbrand** [comment]
> Is the implementation of last_plot() as closure simply a product of older times

I can confirm the implementation is at least [16 years](https://github.com/tidyverse/ggplot2/commit/5461c7f96fff53b7c7117c5175ebeca3ae01d512) old at this time. This doesn't mean anything, but in absence of any other information tips the scales towards the 'this is just old' explanation.

**Reply by yjunechoe** [reply]
Oh wow! And I see theme get/set was also used to be implemented as closure too - interesting 👀
